packages/justiciar/src/rational.ts

```diff
@@ -0,0 +1,171 @@
+/**
+ * A lossless number.
+ */
+export class Rational {
+	protected fractionalValues = new Map<bigint, bigint>()
```

Comment:
yeah that's correct. think of this map as a "holding space" for all the fractional values that comprise this rational number. it can be simplified at any time, but it doesn't have to be done eagerly